Makefile

```diff
@@ -15,6 +15,7 @@ CDEF = -D_FILE_OFFSET_BITS=64 -DRSHIM_LOG_ENABLE=1 -DFUSE_USE_VERSION=30
 CDEF += -DHAVE_RSHIM_NET
 CDEF += -DHAVE_RSHIM_PCIE
 CDEF += -DHAVE_RSHIM_PCIE_LF
+CDEF += -DHAVE_RSHIM_FUSE
```

Comment:
Use "HAVE_RSHIM_FUSE" for Linux and "HAVE_RSHIM_CUSE" for FreeBSD seems a little confusing. 
Linux code actually uses both fuse and cuse. Probably use something else for this difference?

Reply:
ifdef __linux__ and ifdef __FreeBSD__ is not the right thing for this. Do you have any suggestions?

Reply:
I don't have good suggestions either.. Let's make it working first. We could consider improving it later, such as isolating the fs implementation into separate files...